src/overlays/actors/ovl_Bg_Gjyo_Bridge/z_bg_gjyo_bridge.c

```diff
@@ -21,17 +29,94 @@ const ActorInit Bg_Gjyo_Bridge_InitVars = {
     (ActorFunc)BgGjyoBridge_Update,
     (ActorFunc)BgGjyoBridge_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gjyo_Bridge/BgGjyoBridge_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gjyo_Bridge/BgGjyoBridge_Destroy.s")
+InitChainEntry initChain[] = { ICHAIN_F32(unk_F8, 800, ICHAIN_CONTINUE),
+                               ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP) };
+
+extern u32 D_06000600;
+extern u32 D_06000DB8;
+extern u32 D_02002640;
+
+void BgGjyoBridge_Init(BgGjyoBridge* this, GlobalContext* globalCtx) {
+    s32 pad;
+    DynaCollisionContext* dynaCollisionContext;
+    s32 local_c;
+
+    local_c = 0;
+
+    Actor_ProcessInitChain(&this->dyna, &initChain);
+    DynaPolyInfo_SetActorMove(this, 0);
+    DynaPolyInfo_Alloc(&D_06000DB8, &local_c);
+
+    dynaCollisionContext = &globalCtx->colCtx.dyna;
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, dynaCollisionContext, this, local_c);
+
+    if (gSaveContext.event_chk_inf[4] & 0x2000) {
+        this->actionFunc = &func_808787A4;
```

Comment:
This is pretty minor but the & is unnecessary when assigning function pointers, and should usually be omitted. This is relevant in a few other places in this file.
```suggestion
        this->actionFunc = func_808787A4;
```